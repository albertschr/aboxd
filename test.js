const spawn = require("child_process").spawn;

async function runAboxd(str) {
  return new Promise((resolve, reject) => {
    try {
      const process = spawn("./index.js", [str]);

      let output = "";
      process.stdout.on("data", data => {
        output += data.toString();
      });

      process.on("close", () => {
        resolve(output);
      });
    } catch (err) {
      reject(err);
    }
  });
}

it("should render single box", async () => {
  const output = await runAboxd("box");
  expect(output).toMatchSnapshot();
});

it("should render two boxes with lines between ", async () => {
  const output = await runAboxd("box1,box2");
  expect(output).toMatchSnapshot();
});

it("should render two lines of boxes", async () => {
  const output = await runAboxd("box1,box2\nbox3,");
  expect(output).toMatchSnapshot();
});

it("should render three lines of boxes", async () => {
  const output = await runAboxd("box1,box2\nbox3,\nbox4,box5");
  expect(output).toMatchSnapshot();
});

it("should fill in empty slots if a dot is present", async () => {
  const output = await runAboxd("box1,box2\nbox3,.\nbox4,box5");
  expect(output).toMatchSnapshot();
});

it("should tighten up for rows", async () => {
  const output = await runAboxd("box1,box2\n,.\nbox4,box5");
  expect(output).toMatchSnapshot();
});

it("pad lines with fewer cells", async () => {
  const output = await runAboxd("box1,box2\nbox3");
  expect(output).toMatchSnapshot();
});

it("should leave gaps in empty cells", async () => {
  const output = await runAboxd("box1,,box2\nbox3,box4,box5\n,bottom");
  expect(output).toMatchSnapshot();
});
