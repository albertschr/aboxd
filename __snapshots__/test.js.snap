// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pad lines with fewer cells from arg 1`] = `
"
╭──────╮   ╭──────╮
│ box1 │───│ box2 │
╰──────╯   ╰──────╯
   │                  
╭──────╮           
│ box3 │           
╰──────╯           
"
`;

exports[`pad lines with fewer cells from stdin 1`] = `
"
╭──────╮   ╭──────╮
│ box1 │───│ box2 │
╰──────╯   ╰──────╯
   │                  
╭──────╮           
│ box3 │           
╰──────╯           
"
`;

exports[`should fill in empty slots if a dot is present from arg 1`] = `
"
╭──────╮   ╭──────╮
│ box1 │───│ box2 │
╰──────╯   ╰──────╯
   │          │       
╭──────╮      │    
│ box3 │──────┤    
╰──────╯      │    
   │          │       
╭──────╮   ╭──────╮
│ box4 │───│ box5 │
╰──────╯   ╰──────╯
"
`;

exports[`should fill in empty slots if a dot is present from stdin 1`] = `
"
╭──────╮   ╭──────╮
│ box1 │───│ box2 │
╰──────╯   ╰──────╯
   │          │       
╭──────╮      │    
│ box3 │──────┤    
╰──────╯      │    
   │          │       
╭──────╮   ╭──────╮
│ box4 │───│ box5 │
╰──────╯   ╰──────╯
"
`;

exports[`should leave gaps in empty cells from arg 1`] = `
"
╭──────╮                ╭──────╮
│ box1 │                │ box2 │
╰──────╯                ╰──────╯
   │                       │       
╭──────╮   ╭────────╮   ╭──────╮
│ box3 │───│  box4  │───│ box5 │
╰──────╯   ╰────────╯   ╰──────╯
               │                   
           ╭────────╮           
           │ bottom │           
           ╰────────╯           
"
`;

exports[`should leave gaps in empty cells from stdin 1`] = `
"
╭──────╮                ╭──────╮
│ box1 │                │ box2 │
╰──────╯                ╰──────╯
   │                       │       
╭──────╮   ╭────────╮   ╭──────╮
│ box3 │───│  box4  │───│ box5 │
╰──────╯   ╰────────╯   ╰──────╯
               │                   
           ╭────────╮           
           │ bottom │           
           ╰────────╯           
"
`;

exports[`should render single box from arg 1`] = `
"
╭─────╮
│ box │
╰─────╯
"
`;

exports[`should render single box from stdin 1`] = `
"
╭─────╮
│ box │
╰─────╯
"
`;

exports[`should render three lines of boxes from arg 1`] = `
"
╭──────╮   ╭──────╮
│ box1 │───│ box2 │
╰──────╯   ╰──────╯
   │                  
╭──────╮           
│ box3 │           
╰──────╯           
   │                  
╭──────╮   ╭──────╮
│ box4 │───│ box5 │
╰──────╯   ╰──────╯
"
`;

exports[`should render three lines of boxes from stdin 1`] = `
"
╭──────╮   ╭──────╮
│ box1 │───│ box2 │
╰──────╯   ╰──────╯
   │                  
╭──────╮           
│ box3 │           
╰──────╯           
   │                  
╭──────╮   ╭──────╮
│ box4 │───│ box5 │
╰──────╯   ╰──────╯
"
`;

exports[`should render two boxes with lines between  from arg 1`] = `
"
╭──────╮   ╭──────╮
│ box1 │───│ box2 │
╰──────╯   ╰──────╯
"
`;

exports[`should render two boxes with lines between  from stdin 1`] = `
"
╭──────╮   ╭──────╮
│ box1 │───│ box2 │
╰──────╯   ╰──────╯
"
`;

exports[`should render two lines of boxes from arg 1`] = `
"
╭──────╮   ╭──────╮
│ box1 │───│ box2 │
╰──────╯   ╰──────╯
   │                  
╭──────╮           
│ box3 │           
╰──────╯           
"
`;

exports[`should render two lines of boxes from stdin 1`] = `
"
╭──────╮   ╭──────╮
│ box1 │───│ box2 │
╰──────╯   ╰──────╯
   │                  
╭──────╮           
│ box3 │           
╰──────╯           
"
`;

exports[`should tighten up for rows from arg 1`] = `
"
╭──────╮   ╭──────╮
│ box1 │───│ box2 │
╰──────╯   ╰──────╯
              │       
              │    
╭──────╮   ╭──────╮
│ box4 │───│ box5 │
╰──────╯   ╰──────╯
"
`;

exports[`should tighten up for rows from stdin 1`] = `
"
╭──────╮   ╭──────╮
│ box1 │───│ box2 │
╰──────╯   ╰──────╯
              │       
              │    
╭──────╮   ╭──────╮
│ box4 │───│ box5 │
╰──────╯   ╰──────╯
"
`;
